adapters/appnexus.go

```diff
@@ -235,3 +237,42 @@ func NewAppNexusAdapter(config *HTTPAdapterConfig, externalURL string) *AppNexus
 		usersyncInfo: info,
 	}
 }
+
+func CollectImpData(oreq openrtb.BidRequest, req *pbs.PBSRequest, unit pbs.PBSAdUnit) string {
+	data := ""
+
+	//set 0 for web impression, and 1 for mobile impression
+	if oreq.Site != nil {
+		data += "plat:0"
+	} else {
+		data += "plat:1"
+	}
+
+	switch unit.MediaTypes[0] {
```

Comment:
It looks like you've got several independent things going on here: `(site) => plat`, `(mediaType) => fmt`, etc..

This is a bit subjective... but I think these are better split up into helper functions. Google's [C++ styleguide](https://google.github.io/styleguide/cppguide.html#Write_Short_Functions) describes some of the benefits, but I'd add "easier to test" to their list as well. Rather than having to mock a whole `BidRequest`, `PBSRequest`, etc (huge structures with lots of properties), smaller functions would need smaller, fewer arguments.